docs/proposals/scheduling/20220629-schedule-device-in-card-level.md

```diff
@@ -0,0 +1,358 @@
+---
+title: Fine-grained Device Scheduling
+authors:
+- "@buptcozy"
+reviewers:
+- "@eahydra"
+- "@hormes"
+- "@yihuifeng"
+- "@honpey"
+- "@zwzhang0107"
+- "@jasonliu747"
+creation-date: 2022-06-29
+last-updated: 2022-06-29
+status: provisional
+
+---
+
+# Fine-grained Device Scheduling
+
+## Table of Contents
+
+<!--ts-->
+
+* [Fine-grained Device Scheduling](#fine-grained-device-scheduling)
+  * [Summary](#summary)
+  * [Motivation](#motivation-and-user-stories)
+    * [Why Need This Proposal](#why-need-this-proposal)
+      * [NVLink](#nvlink)
+      * [Oversold](#oversold)
+    * [Goals](#goals)
+    * [Non-Goals/Future Work](#non-goalsfuture-work)
+  * [Proposal](#proposal)   
+    * [API](#api) 
+    * [Implementation Details](#implementation-details)
+      * [Scheduler](#scheduler)
+      * [Device reporter](#device-reporter)
+      * [RuntimeProxy](#runtimeProxy)
+    * [Compatibility](#compatibility)
+  * [Unsolved Problems](#unsolved-problems)
+  * [Alternatives](#alternatives)
+  * [Implementation History](#implementation-history)
+  * [References](#references)
+  
+
+<!--te-->
+
+## Summary
+Support schedule devices like gpu\rdma\fpga in card level for better schedule placement. 
+
+## Motivation
+### Why Need This Proposal
+
+This part show why we need device-card-level scheduling. However, each scenario will be a new independent proposal. 
+We list these cases only to express why we need this proposal.
+
+#### NVLink
+![image](/docs/images/nvlink.jpg)
+
+Through the picture we can find that although there's 8 card in A100\V100 node, the data-transport-speed between cards 
+are not same. When a pod need multi-gpu-cards, we have chance to select a better placement with largest data-transport-speed
+in scheduler size. Moreover, when we want a group of pods to use largest data-transport-speed between cards, the scheduler
+should choice the best cards for these pods in-batch. all above need scheduler have ability to schedule based on card level.
+ 
+##### Oversold
+In some cases, we find although some cards free is zero in scheduler account, but we observed that the physical
+utilization of gpu-mem\gpu-core still has buffer, so we may schedule some low-priority jobs to use this buffer.
+When buffer disappear, we should kill these jobs to avoid affecting normal jobs. because the utilization is different
+between cards, the scheduler must have ability to schedule based on card level.
+
+These cases all need scheduler have ability to schedule based on card level.
+
+### Goals
+1.Device reporter will report card level device crd, which will not conflict with kubelet and existed device-plugin.
```

Comment:
Please add Definition Device CRD API and the Resource Definition and Specification in the goals